demos/LowAllocationWebServer/HttpHeaders.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Utf8;
+using LowAllocationServer;
+
+namespace System.Net.Http.Buffered
+{
+    public struct HttpHeaders : IEnumerable<KeyValuePair<Utf8String, Utf8String>>
+    {
+        private readonly ByteSpan _bytes;
+        
+        public HttpHeaders(ByteSpan bytes)
+        {
+            _bytes = bytes;            
+        }                
+
+        public Utf8String this[string headerName]
+        {
+            get
+            {
+                var headers = this.Where(h => h.Key == headerName);
```

Comment:
I thought that would depend on whether the IEnumerator allocates or not. Isn't that true? Because all that does it loop through the enumerator applying the lambda until it return true, in which case it returns the entity. If the enumerator is not allocating, this shouldn't either.

Now, my enumerator is using yield, which I wasn't sure if it would allocate or not. I have a another implementation in a separate branch that does not use yield, although it is a bit more complex. So, I decided to go with the simpler implementation first.